### PR TITLE
Protobufs changes

### LINKED ISSUE #202: more composable/interoperable protos
Hello! I'm working on tooling to generate clients, and it seems that juno doesn't do what others are doing for storing the proto files.

For example,

- [./proto/stargaze](https://github.com/public-awesome/stargaze/tree/main/proto/stargaze)
- [./proto/osmosis](https://github.com/osmosis-labs/osmosis/tree/main/proto/osmosis)
- [./proto/akash](https://github.com/ovrclk/akash/tree/master/proto/akash)
- [./proto/cosmos](https://github.com/cosmos/cosmos-sdk/tree/main/proto)

To make this more interoperable, can we move [./proto/mint/](https://github.com/CosmosContracts/juno/tree/main/proto/mint) so it's `./proto/juno/mint`?

It looks like a pretty small change, but would be HUGE in terms of interoperability.

### PR DESCRIPTION
**Pull request by faddat**
Closes #202 

Incomplete

### CODE REVIEW

**Comment by faddat** [comment]
https://github.com/faddat/archlinux-docker/runs/6502790449?check_suite_focus=true

docker image with buf done

**Comment by faddat** [comment]
@nghuyenthevinh2000 sir, can you please lmk if you think this is now a happy-branch, maybe formally review the total set of changes?

**Comment by nghuyenthevinh2000** [comment]
@faddat 
I just add proto-gen to makefile so that it will be easier for generating proto. The proto generation is fine.

I think the branch is happy :)))

**Comment by faddat** [comment]
@nghuyenthevinh2000 hey great catch!

the .history file is an unintentional artifact created by my work on #207 

can you please remove from this PR, everything except what you think is needed for the protobufs changes?

thanks!

**Comment by faddat** [comment]
@nghuyenthevinh2000 

Do you think that you may be able to solve this issue?

https://github.com/cosmos/cosmos-sdk/issues/12150

I think that you may be most suited to it

**Comment by faddat** [comment]
I think that this is great to get in before a next-release.

**Review by the-frey — APPROVED** [review]
No regressions AFAICT

File: scripts/protocgen2.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+protoc_install_gopulsar() {
+  go install github.com/cosmos/cosmos-proto/cmd/protoc-gen-go-pulsar@latest
+  go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest
+}
+
+protoc_install_gopulsar
+
+echo "Cleaning API directory"
+(cd api; find ./ -type f \( -iname \*.pulsar.go -o -iname \*.pb.go -o -iname \*.cosmos_orm.go -o -iname \*.pb.gw.go \) -delete; find . -empty -type d -delete; cd ..)
```
[the remark this answers is not in the document]
Comment (nghuyenthevinh2000) [reply scripts/protocgen2.sh:13]: the script is used in protocgen.sh but it has been commented out. 

I think Jacob put it there as a placeholder for future modification.

[the remark this answers is not in the document]
Comment (giansalex) [reply scripts/protocgen2.sh:13]: I think it is necessary in cosmos-sdk but not here. Anyway, it is not a problem.

**Review by giansalex — APPROVED** [review]

**Comment by pyramation** [comment]
thanks so much this looks amazing @faddat @nghuyenthevinh2000 @the-frey @giansalex

**Pull request merged**